numerics/quadrature_body.hpp

```diff
@@ -24,35 +34,133 @@ Primitive<std::invoke_result_t<Function, Argument>, Argument> Gauss(
   for (int i = 0; i < points; ++i) {
     Argument const scaled_node = lower_bound + half_width * (nodes[i] + 1);
     // TODO(phl): Consider compensated summation.
-    result += weights[i] * function(scaled_node);
+    result += weights[i] * f(scaled_node);
   }
   return result * half_width;
 }
 
 template<int points, typename Argument, typename Function>
 Primitive<std::invoke_result_t<Function, Argument>, Argument> GaussLegendre(
-    Function const& function,
+    Function const& f,
     Argument const& lower_bound,
     Argument const& upper_bound) {
   static_assert(points < LegendreRoots.size,
                 "No table for Gauss-Legendre with the chosen number of points");
-  return Gauss<points>(function,
+  return Gauss<points>(f,
                        lower_bound,
                        upper_bound,
                        LegendreRoots[points],
                        GaussLegendreWeights[points]);
 }
 
+template<int points, typename Argument, typename Function>
+Primitive<std::invoke_result_t<Function, Argument>, Argument>
+AutomaticClenshawCurtis(Function const& f,
```

Comment:
It would be good to pick a different name, otherwise the `using` makes it visible outside the internal namespace.